content/en/docs/concepts/instrumenting-library.md

```diff
@@ -0,0 +1,270 @@
+
+If you have any feedback or want to add a new conventions - please come and contribute! [Instrumentation Slack](https://cloud-native.slack.com/archives/C01QZFGMLQ7) or [Specification repo](https://github.com/open-telemetry/opentelemetry-specification) are a good place to start!
+
+## OpenTelemetry API
+
+The first step is to take dependency on the OpenTelemetry API package.
+
+OpenTelemetry has [two main modules](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md) - API and SDK.
+OpenTelemetry API is a set of abstractions and not-operational implementations. Unless application brings OpenTelemetry SDK, your instrumentation does nothing and does not impact application performance. **Libraries should only use OpenTelemetry API.**
+
+You may be rightfully concerned about adding new dependencies, here are some considerations to help you decide how to minimize dependency hell issues:
+
+- OpenTelemetry Trace API reached stability in early 2021, it follows [Semantic Versioning 2.0](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md) and we take API stability seriously.
+- When taking dependency, use the earliest stable OpenTelemetry API (1.0.*) and avoid updating it unless you have to use new features.
+- While your instrumentation stabilizes consider shipping it as a separate package: it would never break anyone who don't use it. You can keep it in your repo or contribute to OpenTelemetry so it will ship with other instrumentation packages.
+- Semantic Conventions are [not stable yet](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability), It does not cause any functional issues, but you may need to update your instrumentation once in a while. Having it in a plugin or in OpenTelemetry contrib repo may help keeping conventions up-to-date.
+
+TODO/Open question: ownership and control for library instrumentation contributions
+
+## What to instrument
+
+### When **not** to instrument
+
+Some libraries are a thin clients wrapping network calls. Chances are that OpenTelemetry has auto-instrumentation for the underlying RPC client. In this case, library instrumentation may not be necessary. Don't instrument if:
+
+- you library is a thin proxy on top of documented or self-explanatory APIs
+- AND OpenTelemetry has instrumentation for underlying network calls
+- AND there are no conventions your library should follow
+
+If you're in doubt - don't instrument - you can always do it later when you see a need.
+
+If you choose not to instrument, it may still be useful to provide a way to configure OpenTelemetry handlers for your internal RPC client instance. It's essential in languages that don't support fully automatic instrumentation and still useful in others.

```

Comment:
please add: it may still be a good idea to register the library as instrumented in the opentelemetry registry.https://opentelemetry.io/registry/